src/sagemaker/model.py

```diff
@@ -658,12 +658,7 @@ def compile(
             if target_instance_family == "ml_eia2":
                 pass
             elif target_instance_family.startswith("ml_"):
-                self.image_uri = self._compilation_image_uri(
-                    self.sagemaker_session.boto_region_name,
-                    target_instance_family,
-                    framework,
-                    framework_version,
-                )
+                self.image_uri = job_status.get("InferenceImage", None)
```

Comment:
Is this param expected to be there always? What if this is None? Is the removed method needed?

Reply:
Neo service puts `InferenceImage` in the response according to the [API response](https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_DescribeCompilationJob.html#:~:text=number%2C%0A%20%20%20%22FailureReason%22%3A%20%22string%22%2C-,%22InferenceImage%22,-%3A%20%22string%22%2C%0A%20%20%20%22InputConfig%22%3A%20%7B%20%0A%20%20%20%20%20%20%22DataInputConfig). In the case that the user uses ml instance, the `InferenceImage` will not be None from our service. 
We are making this change so the user can get the latest inference url recommendation.

Is the removed method needed?
Thanks for pointing out. Will delete it